src/core/document.js

```diff
@@ -321,16 +321,7 @@ class Page {
     const savedDict = pageDict.get("Annots");
     pageDict.set("Annots", annotationsArray);
     const buffer = [];
-
-    let transform = null;
-    if (this.xref.encrypt) {
-      transform = this.xref.encrypt.createCipherTransform(
-        this.ref.num,
-        this.ref.gen
-      );
-    }
-
-    await writeObject(this.ref, pageDict, buffer, transform);
+    await writeObject(this.ref, pageDict, buffer, this.ref);
```

Comment:
Given the typo in the last argument, if this passes tests as-is that'd unfortunately suggest that our test-coverage is slightly lacking here.
```suggestion
    await writeObject(this.ref, pageDict, buffer, this.xref);
```

Reply:
I just ran the unit tests locally.
That said this part should be covered by reftests.

Reply:
Here we just write the updated page dict which shouldn't contain any strings, streams, ... so the encryption is useless in this case.